app/main.py

```diff
@@ -546,11 +594,121 @@ def callback_func_start_stop_interval(n_clicks):
 #     return build_alerts_elements(alert_status, alert_metadata)
 
 
+# ----------------------------------------------------------------------------------------------------------------------
+# Callbacks related to alert_screen page
+@app.callback(
+    [
+        Output("core_layout_alert_screen", "children"),
+        Output("core_layout_alert_screen", "style"),
+        Output("last_displayed_event_id", "data"),
+    ],
+    Input("interval-component-alert-screen", "n_intervals"),
+    [
+        State("div_live_alerts_data", "data"),
+        State("last_displayed_event_id", "data"),
+    ],
+)
+def update_alert_screen(n_intervals, live_alerts, last_displayed_event_id):
+    """
+    -- Update elements related to the Alert Screen page when the interval component "alert-screen" is triggered --
+    """
+    if live_alerts is None:
+        style_to_display = no_alert_detected_screen()
+        return (
+            [{}],
+            style_to_display,
+            last_displayed_event_id
+        )
+
+    else:
+        # Fetching the last alert
+        live_alerts = pd.read_json(live_alerts)
+        last_alert = live_alerts.loc[live_alerts["id"].idxmax()]
+        last_event_id = str(last_alert["event_id"])
+
+        # Fetching the URL address of the frame associated with the last alert
+        img_url = api_client.get_media_url(last_alert["media_id"]).json()["url"]
+
+        if last_event_id == last_displayed_event_id:
+            # the alert is related to an event id which has already been displayed
+            # need to send the img_url to the GIF
+            raise PreventUpdate
+        else:
+            # new event, not been displayed yet
+            layout_div, style_to_display = alert_detected_screen(
+                img_url, alert_metadata, last_alert
+            )
+            return layout_div, style_to_display, last_event_id
+
+
+@app.callback(
+    Output("images_url_current_alert", "data"),
+    Input("interval-component-homepage", "n_intervals"),
```

Comment:
Again, I feel like this callback could be merged with one of the two callbacks triggered by `Input("interval-component-homepage", "n_intervals")` above. Typically, I think this one could be merged with the first one that updates the alert data storage component in the sense that we would centralise all API calls in the same callback. What do you think? 🙂

Reply:
Again not sure because these two callbacks are doing very different things. I'd like to keep it so that one function = one task. And anyways I believe that the callback `update_alert_data` will be removed with the webhook